ntoskrnl/mm/balance.c

```diff
@@ -245,15 +251,30 @@ MmTrimUserMemory(ULONG Target, ULONG Priority, PULONG NrFreedPages)
                 /* Nobody accessed this page since the last time we check. Time to clean up */
 
                 Status = MmPageOutPhysicalAddress(CurrentPage);
+                if (NT_SUCCESS(Status))
+                {
+                    if (CurrentPage == FirstPage)
+                    {
+                        FirstPage = 0;
+                    }
+                }
                 // DPRINT1("Paged-out one page: %s\n", NT_SUCCESS(Status) ? "Yes" : "No");
-                (void)Status;
             }
 
             /* Done for this page. */
             Target--;
         }
 
         CurrentPage = MmGetLRUNextUserPage(CurrentPage, TRUE);
+        if (FirstPage == 0)
+        {
+            FirstPage = CurrentPage;
+        }
+        else if (CurrentPage == FirstPage)
+        {
+            DPRINT1("We are back at the start, abort!\n");
```

Comment:
```suggestion
            DPRINT("We are back at the start, abort\n");
```

Reply:
This one is actually worth a DPRINT1, since it really only happens when the balancer struggles to get any new pages. It likely would never show up with the cache patch by Jerome, unless there is too much memory committed.

Reply:
Ok, I restored the 1 in the suggestion.